src/index.ts

```diff
@@ -213,7 +228,12 @@ export async function getIronSession(
   return session as IronSession;
 }
 
-function addToCookies(cookieValue: string, res: ServerResponse) {
+function addToCookies(cookieValue: string, res: ResponseType) {
+  if ("headers" in res) {
```

Comment:
shouldn't we also implement the existing cookie part below? i.e. what if some earlier code already added set-cookie header?

Reply:
There was an issue here actually. According to the [spec](https://fetch.spec.whatwg.org/#headers-class), "a [Headers](https://fetch.spec.whatwg.org/#headers) object cannot represent more than one `Set-Cookie` [header](https://fetch.spec.whatwg.org/#concept-header)." But Next.js' implementation allows this. I have tested that if I write this code:

```
// pages/_middleware.ts

import { NextResponse } from "next/server";
import type { NextFetchEvent, NextRequest } from "next/server";

export function middleware(req: NextRequest, ev: NextFetchEvent) {
  const res = NextResponse.next();
  res.headers.append("set-cookie", "cookie1=value1");
  res.headers.append("set-cookie", "cookie2=value2");
  console.log(res.headers.get("set-cookie"));
  return res;
}
```

This gets logged: `cookie1=value1, cookie2=value2`, and the cookies get set (tested on both latest version of Chrome and Firefox):

![image](https://user-images.githubusercontent.com/40380293/172480948-623bed96-8427-4b3d-a896-80a85aa0c589.png)

![image](https://user-images.githubusercontent.com/40380293/172481110-9ca85feb-c8ff-481b-a235-e6ce7f86ae7f.png)

Reply:
With `res.headers.append`, we don't need to do existing cookie thing. If we had done `res.headers.set`, then it would have been needed. Reference: https://developer.mozilla.org/en-US/docs/Web/API/Headers/set

Reply:
👍